<img src="../../images/Callysto_Notebook-Banner_Top_06.06.18.jpg"/>

In [10]:
%%html

<script>
  function code_toggle() {
    if (code_shown){
      $('div.input').hide('500');
      $('#toggleButton').val('Show Code')
    } else {
      $('div.input').show('500');
      $('#toggleButton').val('Hide Code')
    }
    code_shown = !code_shown
  }
  
  $( document ).ready(function(){
    code_shown=false;
    $('div.input').hide()
  });
</script>
<form action="javascript:code_toggle()"><input type="submit" id="toggleButton" value="Show Code"></form>

# Geographic Progression of The American Revolution


The American Revolution (1775-83) is also known as the American Revolutionary War and the U.S. War of Independence. 

On July 4, 1776, America's Thirteen Colonies declared that they were no longer part of Britain. The British disagreed. A war between Britain and the Thirteen Colonies began as a result of this disagreement. This war was known as the American Revolution or the War of Independence. (From "Our Land and People", Patricia Shields-Ramsay, Douglas Ramsay)


> ![alt text](images/13_colonies.jpg)
>
> The location of the original thirteen English Colonies on the east coast of what is now known as the United States of America. 

In order to understand the origins and progression of the American Revolutionary War, we will look at the locations of various battles during each stage of the war. From this data, we can begin to understand where the battles were taking place, where they moved to, and try and understand the events surrounding them. To begin, we first need to download an open data set from Wikipedia that contains the locations and other information about the battles. The first few rows are shown below

In [11]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
from numpy import nan as Nan
from plotly.offline import init_notebook_mode, iplot
import plotly.graph_objs as go
import plotly.plotly as py
import time
init_notebook_mode(connected=True)

url = 'https://data.opendatasoft.com/explore/dataset/list-of-american-revolutionary-war-battles@public-us/download/?format=csv&timezone=America/Denver&use_labels_for_header=true'
revolution_data  = pd.read_csv(url,sep=';',error_bad_lines=False)
revolution_data.head()

Battle  \
0  Battles of Lexington and Concord   
1         First Siege of Fort Henry   
2          Battle of Freeman's Farm   
3                  Siege of Augusta   
4         Battle of Videau's Bridge   

                                             article                Date  \
0  https://en.wikipedia.org/wiki/Battles_of_Lexin...      April 19, 1775   
1  https://en.wikipedia.org/wiki/Siege_of_Fort_He...                 NaN   
2  https://en.wikipedia.org/wiki/Battles_of_Saratoga  September 19, 1777   
3     https://en.wikipedia.org/wiki/Siege_of_Augusta                 NaN   
4  https://en.wikipedia.org/wiki/Battle_of_Videau...     January 2, 1782   

            State                                            Outcome  \
0   Massachusetts  Patriot victory: British forces raiding Concor...   
1        Virginia                                    Patriot victory   
2        New York  British tactical victory: First of the two Bat...   
3         Georgia                                    Patriot victory   
4  South Carolina                                    British victory   

   Start Date          Victory                Battle location  
0  1775-04-19  Patriot victory  42.4496388889, -71.2308333333  
1         NaN  Patriot victory         40.0638888889, -80.725  
2  1777-09-19              NaN        42.9988888889, -73.6375  
3         NaN  Patriot victory  33.4666666667, -81.9666666667  
4  1782-01-02  British victory                            NaN

From the table above we see that there is a lot of information to digest. In the first column we have the name of the battle, and the second is a link to the Wikipedia page with information about the battle. We also have the date of the battle listed, the state it was located in, the consequences of that battle, and who won. The final column contains the location of the battle, which is represented by two numbers - latitude and longitude. Let's take a brief step back and explore the concepts of latitude and longitude.

## Brief Detour About Geography

**Latitudes and longitudes** are very basic terms in geography. They help us to explain the location or position of something anywhere in the world. Latitudes and longitudes form a grid system on the globe, and by knowing the shape of this grid, we can easily locate any place on Earth.

### Latitude

Lines of latitudes are horizontal systems of measurement that encircle the globe from east to west. Latitudes are sometimes called "parallels," as they are at an equal distance from each other and never meet.

Latitude lines measure how far north or south of the equator something is. The equator is defined as the starting point for measuring latitude - that's why it is marked as 0 degrees latitude. The North Pole is listed at 90 degrees, and the South Pole is listed at -90 degrees. The more "degrees latitude" a location is, the further away it is from the equator. For example, 50 degrees latitude is further from the equator than 20 degrees latitude is. Depending on if a location is above or below the equator, they are listed as "degrees North" or "degrees South", and are given a matching positive (+) or negative (-) symbol. 


Each degree of latitude is approximately 111 kilometers apart (69 miles), although there is some variation as the Earth is not perfectly round. The globe below has the parallels drawn on it. Feel free to spin the globe and zoom in or out on any location. Parallels located south of the Equator are orange, and this part of the Earth is called the Southern Hemisphere. Conversely, parallels located north to the Equator are blue, and this part of the Earth is called the Northern Hemisphere.


In [12]:
contours = []

equator_lon = [-180,-170,-160,-150,-140,-130,-120,-110,-100,-90,-80,-70,-60,-50,-40,-30,-20,-10,0,
               10,20,30,40,50,60,70,80,90,100,110,120,130,140,150,160,170,180]
euator_lat = [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]

#equator
contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= '0° equator',
        name = '0° equator',
        lon = equator_lon,
        lat = euator_lat,
        mode = 'lines',
        line = dict(
            width = 3,
            color = 'green'
        )
    ) )
#north pole
contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= '90° north pole',
        name = '90° north pole',
        lon = [0],
        lat = [90],
        mode = 'markers',
        line = dict(
            width = 2,
            color = 'rgb(50,136,189)'
        )
    ) )
#north hemispere
for i in range (1,9):
    contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= str(10*i)+'° north',
        name = str(10*i)+'° north',
        lon = equator_lon,
        lat = [x+10*i for x in euator_lat],
        mode = 'lines',
        line = dict(
            width = 2,
            color = 'rgb(50,136,189)'
        )
    ) )
    
#south hemispere
for i in range (1,9):
    contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= str(-10*i)+'° south',
        name = str(-10*i)+'° south',
        lon = equator_lon,
        lat = [x-10*i for x in euator_lat],
        mode = 'lines',
        line = dict(
            width = 2,
            color = 'rgb(244,109,67)'
        )
    ) )
    
contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= '-90° south pole',
        name = '-90° south pole',
        lon = [0],
        lat = [-90],
        mode = 'markers',
        line = dict(
            width = 2,
            color = 'rgb(244,109,67)'
        )
    ) )

layout = dict(
        title = 'Parallels over globe<br>(Click and drag to rotate)',        
        geo = dict(
            showland = True,
            showlakes = True,
            showcountries = True,
            showocean = True,
            countrywidth = 0.5,
            lakecolor = 'rgb(255, 255, 255)',
            oceancolor = 'rgb(255, 255, 255)',
            landcolor = 'rgb(212, 212, 212)',
            projection = dict( 
                type = 'orthographic',
                rotation = dict(
                    lon = -100,
                    lat = 40,
                    roll = 0
                )            
            ),

        )
    )
    
fig = dict( data=contours, layout=layout )
iplot( fig, validate=False )

### Questions

Look at the globe. Try rotating it and hover your mouse over different parallels. 
Every tenth parallel is displayed.
If you click on any parallel listing on the right, the latitude line will either appear or disappear on the map.

1. Can you find the Equator? Which continents does it go through?
2. Find Canada. Is it located in the Southern or Northern Hemisphere?
3. Which latitude lines is Canada located between?

### Longitude
In contrast to latitude, longitude lines Lines are vertical systems of measurement that encircle the globe from north to south. Longitude lines converge at poles and are widest at the equator. Longitude lines can also be called "meridian lines". Longitude shows how far a location is east or west of a universal vertical line, called the Prime Meridian. The Prime Meridian line runs through the British Royal Observatory in Greenwich England, from the North Pole to the South Pole. As the vertical starting point for longitude, the Prime Meridian is defined as 0 degrees longitude. Interestingly, for many years, each country had its own prime meridian. In 1884, a group of scientists, navigators, and business owners made Greenwich, England, the official worldwide Prime Meridian.

Longitude lines span 180 degrees both east and west of the Prime Meridian. Longitude lines that are **east** of the prime meridian are positive (+), and lines that are **west** of the Prime Meridian are negative (-). The longitude line that is exactly 180 degrees east and west of the Prime Meridian is known as the International Date Line (it is directly opposite to the Prime Meridian). 

Lines of longitude at every ten degrees are shown on the globe below, try rotating it and hover your mouse over different meridians. If you click on the any meridian listed on the right it will disappear/reappear on the map. On this map, the blue meridians located east of the Prime Meridian are part of the Eastern Hemisphere. The organge meridians located west to the Prime Meridian are part of the Western Hemisphere.

In [13]:
contours = []

mer_lat1 = [90,100,110,120,130,140,150,160,170,180,-180,-170,-160,-150,-140,-130,-120,-110,-100,-90]
mer_lon1 = [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]

mer_lat2 = [-90,-80,-70,-60,-50,-40,-30,-20,-10,0,10,20,30,40,50,60,70,80,90]
mer_lon2 = [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]

#zero meridian
contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= '0° Prime meridian',
        name = '0° Prime meridian',
        lon = mer_lon2,
        lat = mer_lat2,
        mode = 'lines',
        line = dict(
            width = 3,
            color = 'green'
        )
    ) )

contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= '180° International Date Line',
        name = '180° International Date Line',
        lon = mer_lon1,
        lat = mer_lat1,
        mode = 'lines',
        line = dict(
            width = 3,
            color = 'green'
        )
    ) )

#west hemispere
for i in range (1,18):
    contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= str(10*i-180)+'° west',
        name = str(10*i-180)+'° west',
        lat = mer_lat1,
        lon = [x+10*i for x in mer_lon1],
        mode = 'lines',
        line = dict(
            width = 2,
            color = 'rgb(244,109,67)'
        )
    ) )
    
#east hemispere
for i in range (1,18):
    contours.append( dict(
        type = 'scattergeo',
        hoverinfo='text',
        text= str(10*i)+'° east',
        name = str(10*i)+'° east',
        lat = mer_lat2,
        lon = [x+10*i for x in mer_lon2],
        mode = 'lines',
        line = dict(
            width = 2,
            color = 'rgb(50,136,189)'
        )
    ) )
    


layout = dict(
        title = 'Meridians over globe<br>(Click and drag to rotate)',
       # showlegend = False,         
        geo = dict(
            showland = True,
            showlakes = True,
            showcountries = True,
            showocean = True,
            countrywidth = 0.5,
            lakecolor = 'rgb(255, 255, 255)',
            oceancolor = 'rgb(255, 255, 255)',
            landcolor = 'rgb(212, 212, 212)',
            projection = dict( 
                type = 'orthographic',
                rotation = dict(
                    lon = -100,
                    lat = 40,
                    roll = 0
                )            
            ),
    
        )
    )
    
fig = dict( data=contours, layout=layout )
iplot( fig, validate=False )

### Questions



1. Can you locate the Prime Meridian? 
    A. What city in Great Britain does it pass through? 
    B. What other countries does the Prime Meridian pass through?
2. Locate Canada on the globe. Is Canada located in the Eastern or Western Hemisphere?
3. Approximately what lines of latitude is Canada located between?

## Looking at Canada's map
Lets display some of the Canadian cities on the map.

From the map below, if you hover your mouse over different cities, their latitude and longitude locations are  displayed as a pair of numbers: `(longitude, latitude)`. For example, if you hover your mouse over Ottawa, you will see: (-75.7, 45.42). This tells us that that Ottawa is located 75.7 degrees west (longitude) and 45.42 degrees north (latitude).

In [14]:
import plotly.plotly as py
import pandas as pd

df = pd.read_csv('https://raw.githubusercontent.com/plotly/datasets/master/2011_february_us_airport_traffic.csv')
df.head()
name=["Edmonton", "Victoria","Winnipeg","Fredericton","St. John's","Halifax","Toronto",
      #"Charlottetown","Quebec City", 
      "Regina","Whitehorse","Iqaluit","Yellowknife", "Ottawa"]
long=[-113.49,-123.37,-97.14,-66.64, -52.71, -63.57, -79.38,
      #-63.13,-71.21,
      -104.62,-135.06,-68.52,-114.38,-75.70]
latt=[53.54,48.43,49.90,45.97, 47.56, 44.65, 43.65,
      #46.238240,46.813878,
      50.44,60.72,63.75,62.45,45.42]
df['text'] = df['airport'] + '' + df['city'] + ', ' + df['state'] + '' + 'Arrivals: ' + df['cnt'].astype(str)

data = [ dict(
        type = 'scattergeo',
        locationmode = 'north america',
        lon = long,
        lat = latt,
        text = name,
        mode = 'markers+text',
        textposition= 'top',
        marker = dict(
            size = 8,
            opacity = 0.8,
            reversescale = True,
            autocolorscale = False,
            symbol = 'square',
            line = dict(
                width=1,
                color='rgba(102, 102, 102)'
            ),
            color = 'red',
        ))]

layout = dict(
        title = "Some Canada's provincial capitals <br>(Hover for coordinates)",
        colorbar = True,
        geo = dict(
            scope = 'north america',
            showland = True,
            landcolor = "rgb(212, 212, 212)",
            countrycolor = "rgb(255, 255, 255)",
            showlakes = True,
            lakecolor = "rgb(255, 255, 255)",
            showsubunits = False,
            showcountries = True,
            resolution = 50,
            projection = dict(
                type = 'kavrayskiy7',
            ),
            lonaxis = dict(
                showgrid = True,
                gridcolor ='rgb(244,109,67)',
                gridwidth = 2,
                range= [ -140, -50 ],
                dtick = 10
            ),
            lataxis = dict (
                showgrid = True,
                gridcolor ='rgb(50,136,189)',
                gridwidth = 2,
                range= [ 40.0, 80.0 ],
                dtick = 10
            )
        )
    )

fig = dict( data=data, layout=layout )
iplot( fig, validate=False )

### Questions 

1. Are there any negative values of latitude and longitude in Canada? Why or why not?
2. What's the northern most Canadian city on the map? What are its coordinates?
3. Locate Winnipeg and Regina, and note down their coordinates. What do you notice about their latitude values? What does this tell you?
4. Locate both Edmonton and Yellowknife.
    A. What is the difference between their latitudes?
    B. What does the distance between their latitudes tell you?
    C. Given that 1 degree of latitude covers approximately 111 km, calculate the distance between these cities. 

(Answer: (62.45-53.54)*111=989.01km )


## Geographic View of The American Revolution

Now that we have a deeper understanding of what the latitude and longitude values in our Revolutionary War data table mean, lets go back to our dataset of the American Revolutionary battles.

Before we can put this data on a map, we first have to separate our latitude and longitude values from each other. Let's split the **Battle location** column into two: one called "lon" for longitude and once called "lat" for latitude. When separated, these coordinates are much simpler to display on the map. 

We are also interested in looking at the battles by year, so we need to pull the year out of the **Start Date** column and add it to its own column in our data table. NOTE: You will need to remove table entries that are missing values, displayed as `NaN` in the first table, and filter the table so it only shows North American battles. After making those modifications, the first few rows of our will look like this:

In [15]:
revolution_data = revolution_data.dropna(subset=['Battle location'])
revolution_data = revolution_data.dropna(subset=['Start Date'])
revolution_data['lat'] = revolution_data['Battle location'].str.split(', ', expand = True)[0]
revolution_data['lon'] = revolution_data['Battle location'].str.split(', ', expand = True)[1]
revolution_data['Year'] = revolution_data['Start Date'].str[:4]
revolution_data['lat']=pd.to_numeric(revolution_data['lat'], errors='coerce')
revolution_data['lon']=pd.to_numeric(revolution_data['lon'], errors='coerce')
revolution_data=revolution_data[revolution_data['lat']>12]
revolution_data=revolution_data[revolution_data['lat']<60]
revolution_data=revolution_data[revolution_data['lon']>-150]
revolution_data=revolution_data[revolution_data['lon']<-40]
revolution_data=revolution_data[revolution_data['Year'] != '2026']
revolution_data.loc[revolution_data.shape[0]]=[' ',' ',' ',' ',' ',' ','Patriot victory',' ','0','0','1783']

revolution_data.head()

Battle  \
0   Battles of Lexington and Concord   
2           Battle of Freeman's Farm   
5               Mount Hope Bay raids   
6                Capture of Savannah   
12              Capture of Fort Bute   

                                              article                Date  \
0   https://en.wikipedia.org/wiki/Battles_of_Lexin...      April 19, 1775   
2   https://en.wikipedia.org/wiki/Battles_of_Saratoga  September 19, 1777   
5   https://en.wikipedia.org/wiki/Mount_Hope_Bay_r...     May 25-30, 1778   
6   https://en.wikipedia.org/wiki/Capture_of_Savannah   December 29, 1778   
12  https://en.wikipedia.org/wiki/Capture_of_Fort_...   September 7, 1779   

            State                                            Outcome  \
0   Massachusetts  Patriot victory: British forces raiding Concor...   
2        New York  British tactical victory: First of the two Bat...   
5    Rhode Island                                    British victory   
6         Georgia                                    British victory   
12   West Florida                            Patriot-Spanish victory   

    Start Date          Victory                Battle location      lat  \
0   1775-04-19  Patriot victory  42.4496388889, -71.2308333333  42.4496   
2   1777-09-19              NaN        42.9988888889, -73.6375  42.9989   
5   1778-05-25  British victory                41.694, -71.216   41.694   
6   1778-12-29  British victory  32.0508333333, -81.1038888889  32.0508   
12  1779-09-07              NaN  30.3236111111, -91.1369444444  30.3236   

        lon  Year  
0  -71.2308  1775  
2  -73.6375  1777  
5   -71.216  1778  
6  -81.1039  1778  
12 -91.1369  1779

We can now use this table to display the location of each battle on a map of North America. We can also colour the marker for the battle according to who won the battle, and animate those battle by the year that they took place. The year can be controlled by adjusting the slider on the bottom of the map below, or by simply pressing the play button. Look at the map below and observe the progress of battles that took place during the American Civil war. If you hover your mouse over the battle location, you can see the name of the battle, the location, and who won the battle. 

In [16]:
years_from_col = set(revolution_data['Year'])
years_ints = sorted(list(years_from_col))
years = [str(year) for year in years_ints]
revolution_data['Victory'] = revolution_data['Victory'].replace(np.nan, 'Unknown', regex=True)

victories = []

for victory in revolution_data['Victory']:
    if victory not in victories: 
        victories.append(victory)       

figure = {
    'data': [],
    'layout': {},
    'frames': []
}

figure['layout']['sliders'] = {
    'args': [
        'transition', {
            'duration': 400,
            'easing': 'cubic-in-out'
        }
    ],
    'initialValue': '1774',
    'plotlycommand': 'animate',
    'values': years,
    'visible': True
}

custom_colors = {
    'Patriot victory': 'rgb(171, 0, 250)',
    'Unknown': 'rgb(150, 99, 0)',
    'British victory': 'rgb(99, 110, 250)',
}

figure['layout']['updatemenus'] = [
    {
        'buttons': [
            {
                'args': [None, {'frame': {'duration': 500, 'redraw': False},
                         'fromcurrent': True, 'transition': {'duration': 300, 'easing': 'quadratic-in-out'}}],
                'label': 'Play',
                'method': 'animate'
            },
            {
                'args': [[None], {'frame': {'duration': 0, 'redraw': False}, 'mode': 'immediate',
                'transition': {'duration': 0}}],
                'label': 'Pause',
                'method': 'animate'
            }
        ],
        'direction': 'left',
        'pad': {'r': 10, 't': 87},
        'showactive': False,
        'type': 'buttons',
        'x': 0.1,
        'xanchor': 'right',
        'y': 0,
        'yanchor': 'top'
    }
]

sliders_dict = {
    'active': 0,
    'yanchor': 'top',
    'xanchor': 'left',
    'currentvalue': {
        'font': {'size': 20},
        'prefix': 'Year:',
        'visible': True,
        'xanchor': 'right'
    },
    'transition': {'duration': 300, 'easing': 'cubic-in-out'},
    'pad': {'b': 10, 't': 50},
    'len': 0.9,
    'x': 0.1,
    'y': 0,
    'steps': []
}
figure['layout']['title'] = ' American Revolution (1775-83) battles'
figure['layout']['geo']  =  dict(
        scope = 'north america',
        showland = True,
        landcolor = "rgb(212, 212, 212)",
        subunitcolor = "rgb(255, 255, 255)",
        countrycolor = "rgb(255, 255, 255)",
        showlakes = True,
        lakecolor = "rgb(255, 255, 255)",
        showsubunits = True,
        showcountries = True,
        resolution = 50,
        projection = dict(
            type = 'conic conformal',
            rotation = dict(
                lon = -100
            )
        ),
        lonaxis = dict(
            showgrid = True,
            gridwidth = 0.5,
            range= [ -140.0, -55.0 ],
            dtick = 5
        ),
        lataxis = dict (
            showgrid = True,
            gridwidth = 0.5,
            range= [ 20.0, 60.0 ],
            dtick = 5
        )
)
 

year = 1774
for victory in victories:
    dataset_by_year = revolution_data[revolution_data['Year'] == year]
    dataset_by_year_and_vict = dataset_by_year[dataset_by_year['Victory'] == victory]

    data_dict = {
        'type': 'scattergeo',
        'lon': list(dataset_by_year_and_vict['lon']),
        'lat': list(dataset_by_year_and_vict['lat']),
        'mode': 'markers',
        'text': list(dataset_by_year_and_vict['Battle']),
        'marker': {
            'size': 8, 
            'color': custom_colors[victory]
        },
        'name': victory,
        'legendgroup' : 'group1'
    }
    figure['data'].append(data_dict)
    

for year in years:
    frame = {'data': [], 'name': str(year)}
    for victory in victories:
        dataset_by_year = revolution_data[revolution_data['Year'] == year]
        dataset_by_year_and_vict = dataset_by_year[dataset_by_year['Victory'] == victory]

        data_dict = {
            'type': 'scattergeo',
            'lon': list(dataset_by_year_and_vict['lon']),
            'lat': list(dataset_by_year_and_vict['lat']),
            'mode': 'markers',
            'text': list(dataset_by_year_and_vict['Battle']),
            'marker': {
                'size': 8, 
                'color': custom_colors[victory]
            },
            'name': victory,
            'legendgroup' : 'group'
        }
        frame['data'].append(data_dict)

    figure['frames'].append(frame)
    slider_step = {'args': [
        [year],
        {'frame': {'duration': 300, 'redraw': False},
         'mode': 'immediate',
       'transition': {'duration': 300}}
     ],
     'label': year,
     'method': 'animate'}
    sliders_dict['steps'].append(slider_step)

    
figure['layout']['sliders'] = [sliders_dict]
iplot(figure, validate=False)

Using this interactive map, you can observe the progression of the American Revolution in North America in terms of the battles fought each year. Pay particular attention to the battles taking place in Quebec, the Maritime provinces, and along the Saint Lawrence River. 


### Questions
1. Use the slider and hover over the battles on the map above, which battles were fought in what is now Canada?
2. Why did some battles of the American Revolution take place in Quebec? 
3. Choose one of those battles and study it further. How might that battle have shaped Canada as we know it today?



## The Progression of the War
While the map above makes it easy to see where and when various battles were fought during the American Revolution, sometimes it's easier to learn things from graphs and numbers. In the table below, we've drawn the same data from the Wikipedia page, but this time we've counted how many battles took place each year. Summarizing the data in this way allows us to quickly see which years had the most battles.  

In [17]:
import colorlover as cl
results=revolution_data.groupby(['Year']).size().reset_index(name='Count').sort_values(by=['Count'])
results= results[results['Year']!=" "].reset_index(drop=True)
results["Colour"]=['rgb(255,255,255)'] + cl.scales['9']['seq']['Blues'] 
results=results.sort_values(by=['Year'])
#results = results.rename(columns={"Count":"Battles Fought"})
trace0 = go.Table(
  # type = 'table',
  header = dict(
    values = [ "<b>YEAR</b>","<b>NUMBER OF BATTLES</b>",],
    line = dict(color = 'black'),
    fill = dict(color = 'white'),
    align = ['center'],
    font = dict(color = 'black', size = 12)
  ),
  cells = dict(
    values = [results.Year, results.Count],
    line = dict(color = 'rgb(0,0,0)'),
    fill = dict(color = 'rgb(255,255,255)'),
    align = 'center',
    height = 30,
    font = dict(color = 'rgb(0,0,0)', size = 20)
    ))

data = [trace0]

iplot(data)


From the table above, we can see that the year 1777 had the most battles out of the entire war. This is because 1776-1777 was the beginning of the British counter offensive, and British forces made considerable progress against the Americans. 

### Questions
1. 1778 has about half as many battles as 1777, the year of the British counter offensive. What is the reason for this?
2. 1780 has the second most battles in a year. What events lead to this year having as many battles as it did?



## American Revolution Battle Locations

In the figure below we have again used our modified data table to count the locations that had the most battles during the war. In this case, we counted how many battles took place in each state/province/other region throughout the course of the war, and put it on a bar chart. We can easily see which locations had the most battles. The plot below is interactive - if you hover your mouse over a bar, it will display the number of battles that took place, as well as the percentage of battles that took place at that location.

In [18]:
results=revolution_data.groupby(['State']).size().reset_index(name='Count').sort_values(by=['Count'],ascending=False)
results= results[results['State']!=" "].reset_index(drop=True)
results['Percent'] = round(results['Count']*100/sum(results['Count']),2).astype(str) +"%"
trace0 = go.Bar(
    x=results['State'],
    y=results['Count'],
    text=results['Percent'],
    marker=dict(
        color='rgb(158,202,225)',
        line=dict(
            color='rgb(8,48,107)',
            width=1.5,
        )
    ),
    opacity=0.6
)

data = [trace0]
layout = go.Layout(
    title='Number of battles by location',
   # font=dict(size=18),
    margin = dict(
    l= 30,
    r= 30,
    t= 50,
    b= 250
    )
)
fig = go.Figure(data=data, layout=layout)
iplot(fig)

From the plot above, we see that the majority of battles took place in New York, South Carolina and New Jersey. This is another example of taking data that we can easily obtain online to learn more about a subject that may not have been obvious before. 

### Questions
1. What percentage of battles took place in New York, South Carolina and New Jersey?

## Summary

During our lesson about latitude and longitude, and how they describe different locations on the globe, we also learned about how latitude and longitude information can help us learn about The American Revolution. Using latitude and longitude numbers to describe the locations of battles, we were able to visualize the advancement of the American Revolution. We were also able to discover how many battles took place each year, and which states or provinces had the most battles throughout the war. 




<img src="../../images/Callysto_Notebook-Banners_Bottom_06.06.18.jpg"/>